terraform/plan.go

```diff
@@ -61,7 +62,12 @@ func (p *Plan) Context(opts *ContextOpts) (*Context, error) {
 		// Even if we're overriding the state, it should be logically equal
 		// to what's in plan. The only valid change to have made by the time
 		// we get here is to have incremented the serial.
-		return nil, errors.New("plan state and ContextOpts state are not equal")
+		//
+		// Due to the fact that serialization may change the representation of
+		// the state, there is little chance that these aren't actually equal.
+		// Log the error condition for reference, but continue with the state
+		// we have.
+		log.Println("[ERROR] plan state and ContextOpts state are not equal")
```

Comment:
Given how much of a pain it was to get to the bottom of the error here, would it be worth us including some string representation of the state in the log in this case? I guess it might be a bit noisy to dump two whole states, but since it only happens in this exceptional path maybe it's okay? :man_shrugging: